src/uct/ib/dc/base/dc_iface.c

```diff
@@ -132,6 +133,21 @@ static ucs_status_t uct_dc_iface_dci_connect(uct_dc_iface_t *iface, uct_rc_txqp_
     return UCS_OK;
 }
 
+/* restore dc qp from error to rts state */
+ucs_status_t uct_dc_iface_dci_reconnect(uct_dc_iface_t *iface,
```

Comment:
Just to improve readability and consistency, it will complicate `uct_dc_ep_set_failed` function a bit - one more error check with fatal message.